src/components/views/login/InteractiveAuthEntryComponents.js

```diff
@@ -339,6 +348,8 @@ export const MsisdnAuthEntry = React.createClass({
                     errorText: "Token incorrect",
                 });
             }
+        }).catch((e) => {
```

Comment:
this feels weird.

either move `this.props.fail(e)` to here, or replace `catch(...).then(...).catch(...)` with `then(..., ...)`.